resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/DriverServiceFeatureStep.scala

```diff
@@ -38,15 +38,21 @@ private[spark] class DriverServiceFeatureStep(
     s"$DRIVER_HOST_KEY is not supported in Kubernetes mode, as the driver's hostname will be " +
       "managed via a Kubernetes service.")
 
+  def isDigit(char: Char) : Boolean = {
```

Comment:
There's already a JDK method for this anyway, but this isn't the check you want to make.

Reply:
Character.isDigit(preferredServiceName.charAt(0))